src/main/scala/com/gu/inspectorlambda/chiefinspector/ChiefInspector.scala

```diff
@@ -33,6 +33,14 @@ object ChiefInspector extends StrictLogging {
     // Sleeping for 10 seconds to allow for tags propagation
     Thread.sleep(10000)
 
+    // These calls, and their underlying implementations, can be deleted once the old bad targets and
```

Comment:
It seems a shame to be waiting 10 seconds while there's still other work that can be done. These deletes are totally unrelated to the logic that requires a 10 second pause, so it's weird that they are affected by it.

It's ok because this is temporary, but to me this says "this program should be doing concurrency properly". As soon as a `Thread.sleep` is required doing everything one after the other in a blocking call probably isn't the best. What we actually have is a number of independent operations that need a pause in the middle of their steps. This isn't what's described by the code here, and it means we have to start worrying about things like the lambda timeout. `Thread.sleep` will pause the entire program so we're sat idle while there's work to be done.

Scala has a reasonably powerful primitive for describing concurrent programs, in `Future`. It composes nicely, so it would be simple to describe this program's execution properly - a number of independent operations, some of which involve a pause.

Like I said yesterday, it's a bit late now. We're already a ways down the path of this written like a bash script. The issue here is only a temporary one as well. I just offer this as an observation, in particular that when a `Thread.sleep` appears you are dealing with concurrency, so that's probably a good time to introduce language features that make doing that easy. Of course, this also ties in with the testability stuff we chatted about yesterday!

Reply:
I'm not sure we actually could do anything here concurrently though.

The AWS service isn't ACID in any sense for this service, so if we were to do a delete/create without a sleep between, we could get a failure.   It's an eventual consistency problem.

Reply:
I'd hesitate to try to organise the logic to do a bunch of concurrent operations (which may or may not interact), followed by a pause, followed by another bunch.  The problem is that there are restrictions behind the scenes such as not having the same underlying target.

Reply:
Sleeps removed.